manifests/profile/master/postgres.pp

```diff
@@ -0,0 +1,56 @@
+# @summary Aplly this class to an agent running pe-postgresql to collect postgres metrics
+#
+# @param query_interval
+#   How often to run the queries in minutes.  Defaults to 10 minutes.
+#
+# @param postgres_host
+#   The FQDN of the postgres host.  Defaults to the FQDN of the server where the profile is applied
+#
+# @param port
+#   The port that the postgres service listens on.  Defaults to 5432
+define puppet_metrics_dashboard::profile::master::postgres (
+  Variant[String,Tuple[String, Integer]] $postgres_host = $facts['networking']['fqdn'],
+  String[2] $query_interval                             = $puppet_metrics_dashboard::params::pg_query_interval,
+  Integer[1] $port                                      = 5432,
+  ){
+
+  if ! defined(Puppet_metrics_dashboard::Certs['telegraf']) {
+    puppet_metrics_dashboard::certs{'telegraf':
+      notify  => Service['telegraf'],
+      require => Package['telegraf'],
+      before  => Service['telegraf'],
+    }
+  }
+
+  telegraf::input { "pe_postgres_${postgres_host}":
+    plugin_type => 'postgresql_extensible',
+    options     => [{
+      'interval'      => $query_interval,
+      'address'       => "postgres://telegraf@${postgres_host}:${port}/pe-puppetdb?sslmode=require&sslkey=/etc/telegraf/${trusted['certname']}_key.pem&sslcert=/etc/telegraf/${trusted['certname']}_cert.pem&sslrootcert=/etc/telegraf/ca.pem",
```

Comment:
```suggestion
      'address'       => "postgres://telegraf@${facts['networking']['fqdn']}:${port}/pe-puppetdb?sslmode=require&sslkey=/etc/telegraf/${trusted['certname']}_key.pem&sslcert=/etc/telegraf/${trusted['certname']}_cert.pem&sslrootcert=/etc/telegraf/ca.pem",
```

I added the fqdn here because something is making me think its needed instead of simply `localhost`... if that's not the case then my suggestion is to switch to using `localhost`.